sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/LedgerTimeTest.scala

```diff
@@ -33,26 +33,103 @@ class LedgerTimeTest(majorLanguageVersion: LanguageMajorVersion)
 
   private[this] val now = Time.Timestamp.now()
   private[this] val nowP1 = now.addMicros(1)
+  private[this] val nowP2 = now.addMicros(2)
 
   "ledger time primitives" - {
     "ledgerTimeLT" in {
-      val testData = Table[Time.Timestamp, Time.Timestamp, Boolean](
-        ("now", "time", "result"),
-        (nowP1, now, false),
-        (now, now, false),
-        (now, nowP1, true),
+      val testData = Table[Time.Timestamp, Time.Timestamp, Time.Range, Option[Boolean], Time.Range](
+        ("now", "time", "time-boundary", "result", "updated-time-boundary"),
+        // now > time
+        // - lb < ub
+        //   + time < lb
+        (nowP2, now, Time.Range(nowP1, nowP2), Some(false), Time.Range(nowP1, nowP2)),
+        //   + time == lb
+        (nowP2, now, Time.Range(now, nowP1), Some(false), Time.Range(now, nowP1)),
+        //   + time > lb
+        (nowP2, nowP1, Time.Range(now, nowP1), Some(false), Time.Range(nowP1, nowP1)),
+        // - lb == ub
+        //   + time < lb
+        (nowP2, now, Time.Range(nowP1, nowP1), Some(false), Time.Range(nowP1, nowP1)),
+        //   + time == lb
+        (nowP2, now, Time.Range(now, now), Some(false), Time.Range(now, now)),
+        //   + time > lb
+        (nowP2, nowP1, Time.Range(now, now), None, Time.Range(now, now)), // Speedy crash expected
+
+        // now == time
+        // - lb < ub
+        //   + time < lb
+        (now, now, Time.Range(nowP1, nowP2), Some(false), Time.Range(nowP1, nowP2)),
+        //   + time == lb
+        (now, now, Time.Range(now, nowP1), Some(false), Time.Range(now, nowP1)),
+        //   + time > lb
+        (nowP1, nowP1, Time.Range(now, nowP1), Some(false), Time.Range(nowP1, nowP1)),
+        // - lb == ub
+        //   + time < lb
+        (now, now, Time.Range(nowP1, nowP1), Some(false), Time.Range(nowP1, nowP1)),
+        //   + time == lb
+        (now, now, Time.Range(now, now), Some(false), Time.Range(now, now)),
+        //   + time > lb
+        (
+          nowP2,
+          nowP2,
+          Time.Range(nowP1, nowP1),
+          None,
+          Time.Range(nowP1, nowP1),
+        ), // Speedy crash expected
+
+        // now < time
+        // - lb < ub
+        //   + time-1 < ub
+        (now, nowP1, Time.Range(now, nowP1), Some(true), Time.Range(now, now)),
+        //   + time-1 == ub
+        (now, nowP2, Time.Range(now, nowP1), Some(true), Time.Range(now, nowP1)),
+        //   + time-1 > ub
+        (now, Time.Timestamp.MaxValue, Time.Range(now, nowP1), Some(true), Time.Range(now, nowP1)),
```

Comment:
Why no test for `Time.Timestamp.MinValue`?

Reply:
For the case where `now < time`, it is impossible to have a value of `now` that is `< Time.Timestamp.MinValue`.

I'll add in additional tests for the case where `now >= time == Time.Timestamp.MinValue`.